### DISCUSSION TITLE
Option to auto-highlight/focus on file of the active scene in the FileSystem dock

### DISCUSSION THREAD

**Discussion by xdl** [godotengine/godot-proposals#9313] [Editor]
In a lot of IDEs (VsCode, JetBrains - Autoscroll from source https://stackoverflow.com/a/27281201/1010076), the currently focused file would be highlighted in the file explorer pane. This is helpful for getting context on what the nearby files are, navigating to them, and seeing the actual filepath of the current file.

This would be useful in Godot too for the opened scene to be highlighted in the FileSystem dock, particularly when I have multiple identically named scenes (e.g. Zombie.tscn) to identify quickly which variation I'm editing at the moment.

This is a pain-point for me at the moment because with identically named scenes, I don't know the scene path of what I'm currently editing.

## Things tried in the UI that could potentially work (but don't at the moment)

Hovering over a tab in scene editor gives an image preview (no filepath):
![image](https://github.com/godotengine/godot-proposals/assets/4680342/3ac9d21c-5105-4e95-9d12-44d155ea3eda)

Hovering over root node gives the type (no filepath):
![image](https://github.com/godotengine/godot-proposals/assets/4680342/5cd1bd5f-5a90-4e71-a6fc-a31b4b32d49c)

(Hovering over a script gives the filepath but not relevant as I tend to use an external editor for GDScript):
![image](https://github.com/godotengine/godot-proposals/assets/4680342/b47ac585-beea-408b-9c85-9b8e4fd904ce)

No technical specifics on how to implement this but I'm happy to have a look if there's enthusiasm.

**Comment by meisei4** [comment]


**Reply by Calinou** [reply]
@meisei4 Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Comment by tle-oops** [comment]
Hey I made a PR for this: https://github.com/godotengine/godot/pull/109587

**Comment by KoBeWi** [comment]
You can right-click the scene and select Show in FileSystem option.

**Comment by ogapo** [comment]
> You can right-click the scene and select Show in FileSystem option.

That's a bit different from having an editor setting to basically keep it in sync with the scene tab though right? Missing piece here is this probably needs to be on an option not always.

**Reply by AThousandShips** [reply]
If the goal here is to navigate to the file rather than highlighting it then indeed it should be opt-in rather than always on or even on by default

**Comment by Repiteo** [comment]
Implemented by godotengine/godot#109587
